File: Server/src/models/testCase.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const testCaseSchema = new Schema(
    {
        content:{
            type:String,
            required:true,
        },
        createdBy:{
            type: mongoose.Schema.Types.ObjectId,
            required: true
        },
        problemId:{
            type: mongoose.Schema.Types.ObjectId,
            required: true
        },
        approved:{
            type:Boolean,
            default:false,
        }
    }
);
const TestCase = mongoose.models.TestCase || mongoose.model('TestCase', testCaseSchema);
module.exports = TestCase;

File: Golang-backend/models/testCase.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type TestCase struct {
	Id        primitive.ObjectID `json:"_id" bson:"_id"`
	Testcase  *string            `json:"testCase" `
	Output    *string            `json:"output" `
	CreatedBy *string            `json:"createdBy"`
	ProblemId *string            `json:"problemId" `
	Approved  bool               `json:"approved"`
}
